core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -465,286 +467,116 @@ class WorkspaceService(
-      .map(
-        // a workspace may have a single child, if that child is the google project: this is deleted as part of the normal process
-        _.filter(c =>
-          c.resourceTypeName != SamResourceTypeNames.googleProject.value || workspace.googleProjectId.value != c.resourceId
-        )
-      )
-    googleProjectChildren <-
-      samDAO.listResourceChildren(SamResourceTypeNames.googleProject, workspace.googleProjectId.value, ctx)
-    blockingChildren = workspaceChildren.toList ::: googleProjectChildren.toList
-  } yield
-    if (blockingChildren.nonEmpty) {
-      val reports =
-        blockingChildren.map(r => ErrorReport(s"Blocking resource: ${r.resourceTypeName} resource ${r.resourceId}"))
-      throw RawlsExceptionWithErrorReport(
-        ErrorReport(StatusCodes.BadRequest, "Workspace deletion blocked by child resources", reports)
-      )
-    }
-
-  private def deleteWorkspaceInternal(workspaceContext: Workspace,
-                                      maybeMcWorkspace: Option[WorkspaceDescription],
-                                      parentContext: RawlsRequestContext
-  ): Future[WorkspaceDeletionResult] = {
-    if (isAzureMcWorkspace(maybeMcWorkspace)) {
-      return Future.failed(
-        new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, "MC workspaces not supported"))
-      )
+    _ <- requesterPaysSetupService.deleteAllRecordsForWorkspace(workspace)
+    workflowsToAbort <- traceFutureWithParent("gatherWorkflowsToAbortAndSetStatusToAborted", ctx)(_ =>
+      submissionsRepository.getActiveWorkflowsAndSetStatusToAborted(workspace)
+    )
+    // Attempt to abort any running workflows so they don't write any more to the bucket.
+    // Notice that we're kicking off Futures to do the aborts concurrently, but we never collect their results!
+    // This is because there's nothing we can do if Cromwell fails, so we might as well move on and let the
+    // ExecutionContext run the futures whenever
+    aborts = traceFutureWithParent("abortRunningWorkflows", ctx)(_ =>
+      Future.traverse(workflowsToAbort)(wf => executionServiceCluster.abort(wf, ctx.userInfo))
+    )
+    _ <- traceFutureWithParent("deleteFastPassGrantsTransaction", ctx)(childContext =>
+      fastPassServiceConstructor(childContext).removeFastPassGrantsForWorkspace(workspace)
+    )
+    // notify leonardo so it can cleanup any dangling sam resources and other non-cloud state
+    _ <- traceFutureWithParent("notifyLeonardo", ctx)(_ =>
+      leonardoService.cleanupResources(workspace.googleProjectId, workspace.workspaceIdAsUUID, ctx)
+    )
+    // Delete Google Project
+    _ <- traceFutureWithParent("deleteGoogleProject", ctx)(_ => deleteGoogleProject(workspace.googleProjectId))
+    // attempt to delete workspace in WSM, in case thsi is a TDR snapshot - but don't fail on it
+    _ = Try(workspaceManagerDAO.deleteWorkspace(workspace.workspaceIdAsUUID, ctx)).recover {
```

Comment:
This is the same behavior we had before, because we would only fail if it was a `McWorkspace` we were trying to delete, which would never happen, since those are handled by `MultiCloudWorkspaceService`.